File: public/examples/integrations/toolkits/slack_api/get_slack_remote_files_info_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "SlackApi.GetSlackRemoteFilesInfo";

// Start the authorization process
const authResponse = await client.tools.authorize({
    tool_name: TOOL_NAME,
    user_id: USER_ID
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "filter_by_channel_id": "C04567ABC",
  "pagination_cursor": "dXNlcjpVMEc5V0ZYTlo=",
  "maximum_items_to_return": 25,
  "filter_files_from_timestamp": "1693526400",
  "timestamp_filter_end": "1696118400"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/toolkits/slack_api/get_slack_remote_files_info_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "SlackApi.GetSlackRemoteFilesInfo"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'filter_by_channel_id': 'C04567ABCD',
    'pagination_cursor': 'dXNlcjoxMjM0NTY3OA==',
    'maximum_items_to_return': 25,
    'filter_files_from_timestamp': '1691000000',
    'timestamp_filter_end': '1693600000'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
